src/thermo/Phase.cpp

```diff
@@ -126,16 +138,27 @@ size_t Phase::findSpeciesLower(const string& name) const
     return loc;
 }
 
-size_t Phase::speciesIndex(const string& nameStr) const
+size_t Phase::speciesIndex(const string& name) const
 {
-    size_t loc = npos;
+    warn_deprecated("Phase::speciesIndex", "'raise' argument not specified; "
+        "Default behavior will change from returning npos to throwing an exception "
+        "after Cantera 3.2.");
```

Comment:
I think we could reduce some of the churn required by this transition by issuing the deprecation warning only if a nonexistent element/species is requested. This way, any user code that never deals with unknown species/elements (or naively ignores the possibility) could make the transition without ever needing to be modified.